src/coreComponents/constitutive/contact/FrictionlessContact.hpp

```diff
@@ -145,18 +149,38 @@ void FrictionlessContactUpdates::computeTraction( localIndex const k,
                                                   arraySlice1d< real64 const > const & oldDispJump,
                                                   arraySlice1d< real64 const > const & dispJump,
                                                   arraySlice1d< real64 > const & tractionVector,
-                                                  arraySlice2d< real64 > const & dTractionVector_dJump ) const
+                                                  arraySlice2d< real64 > const & dTractionVector_dJump,
+                                                  integer const & fractureState ) const
 {
   GEOSX_UNUSED_VAR( k, oldDispJump );
 
-  tractionVector[0] = dispJump[0] >= 0 ? 0.0 : m_penaltyStiffness * dispJump[0];
+  bool const isOpen = fractureState == extrinsicMeshData::contact::FractureState::Open;
+
+  tractionVector[0] = isOpen ? 0.0 : m_penaltyStiffness * dispJump[0];
   tractionVector[1] = 0.0;
   tractionVector[2] = 0.0;
 
   LvArray::forValuesInSlice( dTractionVector_dJump, []( real64 & val ){ val = 0.0; } );
-  dTractionVector_dJump( 0, 0 ) = dispJump[0] >=0 ? 0.0 : m_penaltyStiffness;
+  dTractionVector_dJump( 0, 0 ) = isOpen ? 0.0 : m_penaltyStiffness;
 }
 
+GEOSX_HOST_DEVICE
+void FrictionlessContactUpdates::updateFractureState( localIndex const k,
+                                                      arraySlice1d< real64 const > const & dispJump,
+                                                      arraySlice1d< real64 const > const & tractionVector,
+                                                      integer & fractureState ) const
+{
+  GEOSX_UNUSED_VAR( k, tractionVector );
+
+  if( dispJump[0] >  -LvArray::NumericLimits< real64 >::epsilon )
```

Comment:
I think that, for non linear problems, since we solve to a tolerance it's a bit too stricht to ask for the jump to be strictly greater than 0.0. For example, especially for nearly closed fractures (with small apertures), we don't really want the state to change if we are just a little under 0.0. I know I tried to do so in the prototype code I had and it can lead to convergence issues.

Reply:
So this is not really an IT constraint but more something about your algorithm.
Maybe do you want to define this quantity as a `dispJumpApertureThreshold`, that can be `const` or specified by the user (or the dev to easily check numerical parameters of convergence)?

Reply:
yes, indeed. Ah, yeah, I can do that. Maybe the numeriLitmits can be its default value.